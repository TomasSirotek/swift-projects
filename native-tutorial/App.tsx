import { StatusBar } from "expo-status-bar";
import { StyleSheet, View, Image } from "react-native";
import Button from "./components/Button";
const PlaceholderImage = require("./assets/images/background-image.png");
import * as ImagePicker from "expo-image-picker";
import {useState} from "react";
import ImageViewer from "./components/ImageViewer";

export default function App() {
  const [selectedImage,setSelectedImage] = useState<string>('')


  const pickImage = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      allowsEditing: true,
      quality: 1,
    });

    !result.canceled
      ? setSelectedImage(result.assets[0].uri)
      : alert("You have cancelled the action");
  };

  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <ImageViewer placeholderImageSource={PlaceholderImage} selectedImage={selectedImage} style={styles.image} />
      </View>
      <View style={styles.footerContainer}>
        <Button theme="primary" label="Choose a photo" onPress={pickImage} />
        <Button label="Use this photo" />
      </View>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#111827",
    alignItems: "center",
  },
  imageContainer: {
    flex: 1,
    paddingTop: 58,
  },
  image: {
    width: 320,
    height: 440,
    borderRadius: 18,
  },
  footerContainer: {
    flex: 1 / 3,
    alignItems: "center",
  },
});
