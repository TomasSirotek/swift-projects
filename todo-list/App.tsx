import {
  KeyboardAvoidingView,
  Platform,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";
import { SafeAreaProvider } from "react-native-safe-area-context";
import Task from "./components/Task";
import { useState } from "react";
import { RootSiblingParent } from "react-native-root-siblings";
import Toast from "react-native-root-toast";

const behaviorSettings = Platform.OS === "ios" ? "padding" : "height";


export default function App() {
  const [tasks, setTasks] = useState<string[]>([]);
  const [text, setText] = useState<string>("");
  
  const handleAddTask = () => {
    // set tasks with new task that was passed in with the text
    if (text === "") {
      // Add a Toast on screen.
      let toast = Toast.show("Please fill the fields", {
        duration: Toast.durations.LONG,
      });

      // You can manually hide the Toast, or it will automatically disappear after a `duration` ms timeout.
      setTimeout(function hideToast() {
        Toast.hide(toast);
      }, 5000);

      return;
    }
    setTasks([...tasks, text]);
    // clear the text
    setText("");
  };

  const deleteTask = (index: number) => {
    let newTasks = [...tasks];
    newTasks.splice(index, 1);
    setTasks(newTasks);
  };

  return (
    <SafeAreaProvider>
      <RootSiblingParent>
        <View style={styles.container}>
          <View style={styles.tasksWrapper}>
            <Text style={styles.sectionTitle}>Today's Tasks</Text>
            <View style={styles.items}>
              {/* This is where the tasks will go! */}
              {tasks.map((task, index) => {
                return (
                  <TouchableOpacity
                    key={index}
                    onPress={() => deleteTask(index)}
                  >
                    <Task text={task} />
                  </TouchableOpacity>
                );
              })}
            </View>
          </View>
        </View>
       

        {/* Writing task */}
        <KeyboardAvoidingView
          behavior={behaviorSettings}
          style={styles.writeTaskWrapper}
        >
          <TextInput
            style={styles.input}
            onChangeText={(text) => setText(text)}
            placeholder={"Write a task"}
          />

          <TouchableOpacity testID="pressCreateBtn"  onPress={() => handleAddTask()}>
            <View style={styles.addWrapper}>
              <Text style={styles.addText}>+</Text>
            </View>
          </TouchableOpacity>
        </KeyboardAvoidingView>
      </RootSiblingParent>
     </SafeAreaProvider> 

  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#E8EAED",
  },
  tasksWrapper: {
    paddingTop: 80,
    paddingHorizontal: 20,
  },
  writeTaskWrapper: {
    position: "absolute",
    bottom: 60,
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: "bold",
  },
  items: {
    marginTop: 30,
  },
  input: {
    paddingVertical: 15,
    paddingHorizontal: 15,
    backgroundColor: "#FFF",
    borderRadius: 60,
    borderColor: "#C0C0C0",
    borderWidth: 1,
    width: 250,
  },
  addWrapper: {
    width: 60,
    height: 60,
    backgroundColor: "#FFF",
    borderRadius: 60,
    justifyContent: "center",
    alignItems: "center",
    borderColor: "#C0C0C0",
    borderWidth: 1,
  },
  addText: {
    fontSize: 25,
  },
});
