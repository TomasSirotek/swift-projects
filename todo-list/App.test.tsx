import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import App from "./App";
import { render, fireEvent } from "@testing-library/react-native";
import { TouchableOpacity } from "react-native";
import { RootSiblingParent } from "react-native-root-siblings";
import Task from "./components/Task";
import Toast from "react-native-root-toast";
import { SafeAreaProvider } from "react-native-safe-area-context";

jest.mock("react-native-safe-area-context", () => ({
  SafeAreaProvider: ({ children }: any) => children,
  useSafeAreaInsets: () => ({
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
  }),
}));

describe("<App />", () => {
  it("renders correctly", () => {
    render(<App />);
  });

  it("should match snapshot", () => {
    const { toJSON } = render(
      <SafeAreaProvider>
        <RootSiblingParent>
          <App />
        </RootSiblingParent>
      </SafeAreaProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should add a task when the add button is pressed", () => {
    const { getByTestId, getByPlaceholderText, queryByText } = render(
      <SafeAreaProvider>
        <RootSiblingParent>
          <App />
        </RootSiblingParent>
      </SafeAreaProvider>
    );

    // Input task text
    const taskInput = getByPlaceholderText("Write a task");
    fireEvent.changeText(taskInput, "New Task");

    // Click the "Add" button
    const addButton = getByTestId("pressCreateBtn");
    fireEvent.press(addButton);

    // Verify that the task has been added
    expect(queryByText("New Task")).toBeTruthy();
  });

  it("should not add a task when the input field is empty", () => {
    const { getByTestId, queryByText } = render(<App />);

    // Click the "Add" button without entering any task text
    const addButton = getByTestId("pressCreateBtn");
    fireEvent.press(addButton);

    // Verify that no task is added
    expect(queryByText("")).toBeNull();
  });

  it("should delete a task when it is pressed", () => {
    const { getByTestId, getByText, queryByText, getByPlaceholderText } =
      render(<App />);

    // Add a task
    const taskInput = getByPlaceholderText("Write a task");
    fireEvent.changeText(taskInput, "Task to be deleted");
    const addButton = getByTestId("pressCreateBtn");
    fireEvent.press(addButton);

    // Verify that the task is added
    expect(queryByText("Task to be deleted")).toBeTruthy();

    // Click the task to delete it
    const taskToDelete = getByText("Task to be deleted");
    fireEvent.press(taskToDelete);

    // Verify that the task has been deleted
    expect(queryByText("Task to be deleted")).toBeNull();
  });
});

it("should delete the correct task when the delete button is pressed", () => {
  const { getByTestId, getByText, queryByText, getByPlaceholderText } = render(
    <App />
  );

  // Add two tasks
  const taskInput = getByPlaceholderText("Write a task");
  fireEvent.changeText(taskInput, "Task 1");
  const addButton = getByTestId("pressCreateBtn");
  fireEvent.press(addButton);
  fireEvent.changeText(taskInput, "Task 2");
  fireEvent.press(addButton);

  // Verify that the tasks are added
  expect(queryByText("Task 1")).toBeTruthy();
  expect(queryByText("Task 2")).toBeTruthy();

  // // Click the delete button for the first task
  const deleteButton = getByText("Task 1").findByType(Task);
  fireEvent.press(deleteButton);
  // // Verify that the first task has been deleted
  // expect(queryByText("Task 1")).toBeNull();
  

// it("should not add a task when the input field contains only whitespace", () => {
//   const { getByTestId, queryByText, getByPlaceholderText } = render(<App />);

//   // Input whitespace
//   const taskInput = getByPlaceholderText("Write a task");
//   fireEvent.changeText(taskInput, "   ");

//   // Click the "Add" button
//   const addButton = getByTestId("pressCreateBtn");
//   fireEvent.press(addButton);

//   // Verify that no task is added
//   expect(queryByText("")).toBeNull();
});
