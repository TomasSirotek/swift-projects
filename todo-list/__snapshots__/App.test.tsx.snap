// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#E8EAED",
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "paddingHorizontal": 20,
          "paddingTop": 80,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Today's Tasks
      </Text>
      <View
        style={
          {
            "marginTop": 30,
          }
        }
      />
    </View>
  </View>,
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "bottom": 60,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "position": "absolute",
          "width": "100%",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Write a task"
      style={
        {
          "backgroundColor": "#FFF",
          "borderColor": "#C0C0C0",
          "borderRadius": 60,
          "borderWidth": 1,
          "paddingHorizontal": 15,
          "paddingVertical": 15,
          "width": 250,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="pressCreateBtn"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderColor": "#C0C0C0",
            "borderRadius": 60,
            "borderWidth": 1,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 25,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>,
]
`;
